handler/core/src/main/java/io/knotx/fragments/task/factory/node/NodeWithMetadata.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.task.factory.node;
+
+import io.knotx.fragments.engine.graph.Node;
+import io.knotx.fragments.handler.api.domain.FragmentResult;
+import io.vertx.core.json.JsonObject;
+import java.util.Map;
+
+public abstract class NodeWithMetadata implements Node {
```

Comment:
I'm skeptical about the infrastructural setup we introduce in here.
A `Node` becomes responsible not only for the execution and transition, but also collecting metadata.
I think this responsibility should be moved to the engine that executes the graph to collect those.